addon/model-data.js

```diff
@@ -696,9 +727,10 @@ export default class M3ModelData {
   }
 
   _notifyRecordProperties(changedKeys) {
-    if (this._embeddedInternalModel) {
+    if (this._embeddedInternalModel && this._embeddedInternalModel.record) {
```

Comment:
Why the conjunction?

When is it valid to have an embedded internal model with no record?

Reply:
When we do `pushData` for the embedded model, it will attempt to notify the records, but `record` field of `_embeddedInternalModel` is assigned only after `pushData` has been called. If we re-order these calls, we can get rid of the additional check for record on the embedded internal model.